src/metabase/models/user.clj

```diff
@@ -2,6 +2,7 @@
   (:require
    [clojure.data :as data]
    [clojure.string :as str]
+   [metabase-enterprise.sso.integrations.sso-settings :as sso-settings]
```

Comment:
You're getting a build error on this PR, I think because you're requiring an EE namespace here and EE code isn't guaranteed to be available.

I think the right approach would be actually to change that setting into a multi-setting using the `define-multi-setting` macro. That allows you to dispatch to two separate setting implementations depending on whether EE functionality is enabled. Take a look at the implementation of `google-auth-auto-create-accounts-domain`; you can probably do something similar and put a multi-setting definition in `metabase.integrations.common`, and just make the OSS implementation non-settable and default it to `true`

Reply:
Ah I see, I only tested the code with the ee flag turned on. If I understand correctly, would I be placing both the ee and oss implementations (`define-multi-setting-impl`) also in `metabase.integrations.common`?

Reply:
I think you can put the base definition and the OSS implementation in that namespace and keep the EE one where it is. EE code needs to be in the `enterprise/` directory so that it's protected by a more restrictive license.